docs/arch/02-core-concepts.md

```diff
@@ -0,0 +1,693 @@
+- `thv client list` - Show available clients
+- `thv client setup <client>` - Configure client
+- `thv client status` - Show client status
+
+**Implementation:**
+- Client types: `pkg/client/types.go`
+- Manager: `pkg/client/manager.go`
+
+**Related concepts:** Workload, Group
+
+## Verbs (Actions)
+
+### Deploy
+
+**Deploy** creates and starts a workload with all its components.
+
+**For containers:**
+1. Create container with image
+2. Configure networking and ports
+3. Apply permission profile
+4. Start container
+5. Attach streams (if stdio)
+6. Start proxy
+7. Apply middleware
+8. Update state
+
+**For remote servers:**
+1. Validate remote URL
+2. Start proxy
+3. Configure authentication (if needed)
+4. Apply middleware
+4. Update state
+
+**Commands:**
+- `thv run <image|url>` - Deploy and start
+- `thv run --from-config <file>` - Deploy from config
+
+**Implementation:**
+- CLI: `cmd/thv/app/run.go`
+- Workloads: `pkg/workloads/manager.go`
+- Runtime: `pkg/container/runtime/types.go`
+
+**Related concepts:** Workload, Runtime, Transport
+
+### Proxy
+
+**Proxy** forwards MCP traffic between clients and servers while applying middleware.
+
+**Proxy types:**
+- **Transparent**: Forwards HTTP without parsing
+- **Protocol-specific**: Parses and translates messages
+
+**Proxy operations:**
+1. Start HTTP server on proxy port
+2. Apply middleware chain to requests
+3. Forward to destination (container or remote)
+4. Return responses to clients
+5. Track sessions
+6. Expose telemetry and health endpoints
+
+**Implementation:**
+- Transparent: `pkg/transport/proxy/transparent/transparent_proxy.go`
+- SSE: `pkg/transport/proxy/httpsse/http_proxy.go`
+- Streamable: `pkg/transport/proxy/streamable/streamable_proxy.go`
+
+**Related concepts:** Transport, Middleware, Session
+
+### Attach
+
+**Attach** connects to a container's stdin/stdout streams for stdio transport.
+
+**Attach process:**
+1. Container must be running
+2. Request attach from runtime
+3. Receive stdin (`WriteCloser`) and stdout (`ReadCloser`)
+4. Start message processing goroutines
+5. Read JSON-RPC from stdout
+6. Write JSON-RPC to stdin
+
+**Framing:**
+- Newline-delimited JSON-RPC messages
+- Each message ends with `\n`
+
+**Implementation:**
+- Transport: `pkg/transport/stdio.go`
+- Runtime interface: `pkg/container/runtime/types.go`
+
+**Related concepts:** Stdio Transport, Runtime
+
+### Parse
+
+**Parse** extracts structured information from JSON-RPC MCP messages for middleware processing.
+
+**Parsing includes:**
+- Message type (request, response, notification)
+- Method name (e.g., `tools/call`, `resources/read`)
+- Request ID
+- Parameters
+- Resource ID (for resource operations)
+- Arguments (for tool calls)
+
+**Parsed data stored in context:**
+- Available to downstream middleware
+- Used by authorization for policy evaluation
+- Used by audit for event logging
+
+**Implementation:**
+- Parser middleware: `pkg/mcp/parser.go`
+- Parsed type: `pkg/mcp/types.go`
```

Comment:
File path reference is incorrect: `pkg/mcp/types.go` does not exist.

MCP types are defined directly in `pkg/mcp/parser.go`.

Consider updating to reference `pkg/mcp/parser.go`.

Evidence: Verified via codebase exploration